File: windows/PomodoroTimerApp/PomodoroTimers/States/RunningState.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Timers;

namespace PomodoroTimerApp.PomodoroTimers.States
{
    internal class RunningState : State
    {
        public RunningState(PomodoroTimer pomodoroTimer) : base(pomodoroTimer)
        {
        }

        public override void Completed()
        {
            _timer.ChangeState(new ReadyState(_timer));
            _timer.HandleTimerCompletion();
            
        }

        public override bool ActivityPause()
        {
            _timer.Pause();
            _timer.ChangeState(new PausedState(_timer));
            return true;
        }

        public override bool ActivityResume()
        {
            return false;
        }


        public override void StartPauseResume()
        {
            _timer.Pause();
            _timer.ChangeState(new PausedState(_timer));
        }
        public override void Stop()
        {
            _timer.Stop();
            _timer.ChangeState(new ReadyState(_timer));
        }
    }
}

File: macos/Sources/PomodoroCore/PomodoroTimers/States/RunningState.swift
import Foundation

struct RunningState: TimerState {
    let kind: TimerStateKind = .running

    func startPauseResume(_ timer: PomodoroTimer) {
        timer.pause()
        timer.change(to: PausedState())
    }

    func stop(_ timer: PomodoroTimer) {
        timer.stop()
        timer.change(to: ReadyState())
    }

    /// Order matters and matches the C#: transition to Ready FIRST, then fire completion.
    /// `handleTimerCompletion` synchronously drives the work->break cycle transition, which
    /// tears this timer down; firing it while still marked Running would be observably wrong.
    func completed(_ timer: PomodoroTimer) {
        timer.change(to: ReadyState())
        timer.handleTimerCompletion()
    }

    func activityPause(_ timer: PomodoroTimer) -> Bool {
        timer.pause()
        timer.change(to: PausedState())
        return true
    }

    func activityResume(_ timer: PomodoroTimer) -> Bool { false }
}
